engine/env/spark/src/main/java/org/datacleaner/spark/ApplicationDriver.java

```diff
@@ -159,34 +174,42 @@ public SparkLauncher createSparkLauncher(File hadoopConfDir, String configuratio
         for (final String jar : jars) {
             sparkLauncher.addJar(jar);
         }
+
         sparkLauncher.setMainClass(Main.class.getName());
+        sparkLauncher.setConf("spark.serializer", "org.apache.spark.serializer.JavaSerializer");
 
         // the primary jar is always the first argument
         sparkLauncher.addAppArgs(primaryJar.get());
 
-        sparkLauncher.addAppArgs(toHdfsPath(configurationHdfsPath));
-        sparkLauncher.addAppArgs(toHdfsPath(jobHdfsPath));
+        sparkLauncher.addAppArgs(toHadoopPath(configurationHdfsPath));
+        sparkLauncher.addAppArgs(toHadoopPath(jobHdfsPath));
+
+        if (!StringUtils.isNullOrEmpty(resultHdfsPath)) {
+            Properties properties = new Properties();
+            properties.setProperty("datacleaner.result.hdfs.path", resultHdfsPath);
+            File tempFile = File.createTempFile("job-", ".properties");
+            properties.store(new FileWriter(tempFile), "DataCleaner Spark runner properties");
+            final URI uri = copyFileToHdfs(tempFile,
+                    _fileSystem.getHomeDirectory().toUri().resolve("temp/" + tempFile).toString());
```

Comment:
Nice touch with the home-dir! We should consider using that more ... But not a matter for this PR ;-)